### PR TITLE
fix: push an empty index when deleting the last referrer with SkipReferrersGC

### PR DESCRIPTION
**Pull request by Wwwsylvia**
Fix: #592

### CODE REVIEW

File: registry/remote/repository.go [hunk 1]
```diff
@@ -1453,7 +1453,11 @@ func (s *manifestStore) updateReferrersIndex(ctx context.Context, subject ocispe
 		}
 
 		// 3. push the updated referrers list using referrers tag schema
-		if len(updatedReferrers) > 0 {
+		if len(updatedReferrers) > 0 || s.repo.SkipReferrersGC {
+			// push a new index when:
+			// 1. updatedReferrers is not empty
+			// 2. OR referrers GC is skipped, in this case the old index
+			//    won't get deleted and an empty index should be pushed
```
Comment (shizhMSFT) [inline registry/remote/repository.go:1460]: This block of comment is a bit confusing. You may reword it to be more natural.

Comment (Wwwsylvia) [reply registry/remote/repository.go]: Reworded

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by TerryHowe — APPROVED** [review]
/lgtm

File: registry/remote/repository.go [hunk 2]
```diff
@@ -1463,14 +1468,15 @@ func (s *manifestStore) updateReferrersIndex(ctx context.Context, subject ocispe
...
-		// 4. delete the dangling original referrers index
-		if !skipDelete {
-			if err := s.repo.delete(ctx, oldIndexDesc, true); err != nil {
-				return &ReferrersError{
-					Op:      opDeleteReferrersIndex,
-					Err:     fmt.Errorf("failed to delete dangling referrers index %s for referrers tag %s: %w", oldIndexDesc.Digest.String(), referrersTag, err),
-					Subject: subject,
-				}
+		// 4. delete the dangling original referrers index, if applicable
+		if s.repo.SkipReferrersGC || noOldIndex {
```
Comment (qweeah) [inline registry/remote/repository.go:1472]: Shouldn't `noOldIndex` already be implied by `len(oldReferrers) == 0`?

Comment (Wwwsylvia) [reply registry/remote/repository.go:1472]: No. Consider the following cases:
1. Old referrers index exists and contains at least one referrers: `noOldIndex == false`, `len(oldReferrers) > 0`
2. Old referrers index exists but contains no referrer: `noOldIndex == false`, `len(oldReferrers) == 0`
3. Old referrers index does not exist (server returns 404):  `noOldIndex == true`, `len(oldReferrers) == 0`

We skip deleting the old index when `skipReferrersGC == true` or the old index is not present (case 3).

Comment (qweeah) [reply registry/remote/repository.go:1472]: How about `s.repo.SkipReferrersGC || content.Equal(oldIndexDesc, ocispec.Descriptor{})`

Comment (qweeah) [reply registry/remote/repository.go:1472]: The main purpose of this comment is: whether an old index exists is already implied by returning value of function `referrersFromIndex`.

Comment (Wwwsylvia) [reply registry/remote/repository.go:1472]: The suggested way does have the same effect as the current code. But I think having `noOldIndex` is more readable. Otherwise we will need extra comment to explain what does `content.Equal(oldIndexDesc, ocispec.Descriptor{})` mean.

Comment (Wwwsylvia) [reply registry/remote/repository.go]: Refactored a little bit.

**Review by shizhMSFT — APPROVED** [review]
LGTM
